### ISSUE TITLE
(deprecated api usage)

### ISSUE THREAD

**Issue by aduplat** [goldendict/goldendict#1240]
QSingleApplication API is deprecated by Qt and being maintained adhoc by persons external to Qt.
--
For this reason I cannot install goldendict in my distro: Funtoo.

It is possible to avoid that dependency?

Thank you,

**Comment by vedgy** [comment]
QtSingleApplication is bundled with goldendict source code here: https://github.com/goldendict/goldendict/tree/master/qtsingleapplication. It is not an external dependency. So goldendict can be built without a system QtSingleApplication package on most distributions. Did you get a compilation error?

**Comment by aduplat** [comment]
Hi,
Sorry for the late response.  I'm trying to install the package from Gentoo repositories in Funtoo. 

```
sudo emerge -av goldendict

These are the packages that would be merged, in order:

Calculating dependencies... done!

emerge: there are no ebuilds to satisfy "dev-qt/qtsingleapplication[qt5(+),X]".
(dependency required by "app-text/goldendict-1.5.0_rc2-r2::text-kit" [ebuild])
(dependency required by "goldendict" [argument])
e no ebuilds to satisfy "dev-qt/qtsingleapplication[qt5(+),X]".
(dependency required by "app-text/goldendict-1.5.0_rc2-r2::text-kit" [ebuild])
(dependency required by "goldendict" [argument])

```

**Comment by aduplat** [comment]
I will try your suggestion. Thank you.

**Comment by vedgy** [comment]
Looks like you'll have to edit everything related to QtSingleApplication out of the Gentoo ebuild and get closer to the recommended upstream build process.

**Comment by aduplat** [comment]
I commented your point to the person helping to fix this issue in Funtoo, and I received this answer: 

`Compile w/o QtSingleApplication failed as there were numerous references to QWebKit in code. This is deprecated with QWebEngine as its replacement (see porting page here: https://doc.qt.io/qt-5/qtwebenginewidgets-qtwebkitportingguide.html). This code is not well maintained and needs quite a bit of refactoring to be updated with current libraries. I don't have time to do this and IMHO, upstream dev should be doing this kind of heavy lifting as part of development/maintenance. `
---
Thank you for your time addressing this issue.

I have used Goldendict for years as a foreign language instructor, but I don't have the technical knowledge to understand how to fix this.

Thank you,

**Comment by vedgy** [comment]
Porting Goldendict from Qt WebKit to Qt WebEngine is a difficult and time-consuming task. I doubt it will happen anytime soon. See the comment from the only active maintainer of Goldendict in #1081.

If there is no qtwebkit package in Funtoo similar to this [Gentoo package](https://packages.gentoo.org/packages/dev-qt/qtwebkit), then you'll have to build Qt WebKit manually (you may find some tips for this in #1086). If there *is* a qtwebkit package in Funtoo, then editing the ebuild should be simple.

> This code is not well maintained and needs quite a bit of refactoring to be updated with current libraries.

Goldendict compiles just fine on Gentoo and Arch Linux with current libraries, but these current libraries include Qt WebKit that is still used by some applications.

**Comment by vedgy** [comment]
On second thought, if there is no qtwebkit package in Funtoo, adapting this package from Gentoo should be much easier than building Qt WebKit manually. So that's what you can ask from the person helping you fix the issue in Funtoo.

**Comment by xiaoyifang** [comment]
#1400 for reference

**Comment by data-man** [comment]
@xiaoyifang 
Something is wrong.
E.g. Debian/Devuan has [libsingleapplication-dev](https://github.com/itay-grudev/SingleApplication) but GD uses another sources.

**Comment by vedgy** [comment]
@data-man, GD bundles its own copy of QtSingleApplication. The repository you linked to is an alternative incompatible implementation.

**Comment by data-man** [comment]
@vedgy 
> GD bundles its own copy of QtSingleApplication.

Yes, and it's also outdated: https://github.com/qtproject/qt-solutions

> The repository you linked to is an alternative incompatible implementation.

I know :).
> but GD uses another sources.
